pkg/pgmodel/metrics/database/metrics.go

```diff
@@ -0,0 +1,188 @@
+			},
+		),
+		query: `select count(*)::bigint from _timescaledb_catalog.chunk`,
+	},
+	{
+		metric: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "chunks_compressed_count",
+				Help:      "Total number of compressed chunks in TimescaleDB currently.",
+			},
+		),
+		query: `select count(*)::bigint from _timescaledb_catalog.chunk where dropped=false and compressed_chunk_id is not null`,
+	}, {
+		metric: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "compression_status",
+				Help:      "Compression status in TimescaleDB.",
+			},
+		),
+		query: `select (case when (value = 'true') then 1 else 0 end) from _prom_catalog.default where key = 'metric_compression'`,
+	}, {
+		metric: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "worker_count",
+				Help:      "Number of TimescaleDB background workers.",
+			},
+		),
+		query: `select current_setting('timescaledb.max_background_workers')::BIGINT`,
+	}, {
+		metric: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "worker_maintenance_job",
+				Help:      "Number of Promscale maintenance workers.",
+			},
+		),
+		query: `select count(*) from timescaledb_information.jobs where proc_name = 'execute_maintenance_job'`,
+	}, {
+		metric: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "worker_maintenance_job_failed",
+				Help:      "Number of Promscale maintenance workers.",
+			},
+		),
+		query: `select count(stats.last_run_status)
+	from timescaledb_information.job_stats stats
+inner join
+	timescaledb_information.jobs jobs
+		on jobs.job_id = stats.job_id
+	where jobs.proc_name = 'execute_maintenance_job' and stats.last_run_status = 'Failed'`,
+	}, {
+		metric: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "worker_maintenance_job_start_timestamp_seconds",
+				Help:      "Timestamp in unix seconds for last successful execution of Promscale maintenance job.",
+			},
+		),
+		query: `SELECT extract(
+	epoch FROM (SELECT COALESCE(
+		(SELECT last_run_started_at AS job_running_since
+			FROM   timescaledb_information.job_stats WHERE  last_run_started_at > last_successful_finish
+				AND last_run_status = 'Success'
+		),
+		CURRENT_TIMESTAMP
+	)))::BIGINT`,
+	}, {
+		metric: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "metric_count",
+				Help:      "Total number of metrics in the database.",
+			},
+		),
+		query: `select count(*)::bigint from _prom_catalog.metric`,
+	},
+}
+
+var metricMux sync.Mutex
```

Comment:
I think we should get rid of this. Maintaining this lock will not help with modifying metric from some other place.